stdlib/http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java

```diff
@@ -287,7 +287,7 @@ private static Object getRecord(BType entityBodyType, Object bjson) {
      */
     private static Object getBJsonValue(ObjectValue inRequestEntity) {
         Object bjson = EntityBodyHandler.constructJsonDataSource(inRequestEntity);
-        EntityBodyHandler.addMessageDataSource(inRequestEntity, bjson);
+        EntityBodyHandler.addMessageDataSource(inRequestEntity, bjson, true);
```

Comment:
@riyafa Instead of passing a flag can't we come up with two different methods... passing a flag itself says a particular method is doing more than one thing.

Reply:
Do you mean a methods as `addJsonMessageDataSource` and `addMessageDataSource`

Reply:
Still we would be setting the native data flag. Would it be fine?
 https://github.com/ballerina-platform/ballerina-lang/pull/17978/files#diff-19f44dc0b5d99966731af874b737d17fR115

Reply:
Exactly! That latter is fine.